src/vstart.sh

```diff
@@ -736,6 +742,57 @@ $extra_conf
 EOF
 }
 
+write_logrotate_conf() {
+    mon_logs=$(pwd)"/out/mon.*.log"
+    osd_logs=$(pwd)"/out/osd.*.log"
+    mgr_logs=$(pwd)"/out/mgr.*.log"
+    mds_logs=$(pwd)"/out/mds.*.log"
+
+    mon_pids=
+    osd_pids=$(ps -e | grep ceph-osd | awk '{print $1}' | tr '\n' ' ')
+    mds_pids=$(ps -e | grep ceph-mds | awk '{print $1}' | tr '\n' ' ')
+    mgr_pids=$(ps -e | grep ceph-mgr | awk '{print $1}' | tr '\n' ' ')
+
+    cat << EOF
+$mon_logs
+$osd_logs
+$mgr_logs
+$mds_logs
+{
+    copytruncate
+    missingok
+    size 2G
+    rotate 20
+    prerotate
+        kill -s stop \$(ps -e | grep ceph-mon | awk '{print \$1}' | tr '\n' ' ')
+        kill -s stop \$(ps -e | grep ceph-osd | awk '{print \$1}' | tr '\n' ' ')
+        kill -s stop \$(ps -e | grep ceph-mgr | awk '{print \$1}' | tr '\n' ' ')
+        kill -s stop \$(ps -e | grep ceph-mds | awk '{print \$1}' | tr '\n' ' ')
+    endscript
+    postrotate
+        kill -s cont \$(ps -e | grep ceph-mon | awk '{print \$1}' | tr '\n' ' ')
+        kill -s cont \$(ps -e | grep ceph-osd | awk '{print \$1}' | tr '\n' ' ')
+        kill -s cont \$(ps -e | grep ceph-mgr | awk '{print \$1}' | tr '\n' ' ')
+        kill -s cont \$(ps -e | grep ceph-mds | awk '{print \$1}' | tr '\n' ' ')
+    endscript
+}
+EOF
+}
+
+init_logrotate() {
+    logrotate_conf_path=$(pwd)"/logrotate.conf"
+    logrotate_state_path=$(pwd)"/logrotate.state"
+
+    if ! test -a $logrotate_conf_path; then
+        if test -a $logrotate_state_path; then
+            rm -f $logrotate_state_path
+        fi
+        write_logrotate_conf > $logrotate_conf_path
+    fi
+
+    logrotate $logrotate_conf_path -s $logrotate_state_path
```

Comment:
Is it necessary to run this before hand?

I'm wondering if it makes sense to always produce a logrotate.conf|state. The logrotate process is not persistent so there's no reason to not create the conf file.

Reply:
No, not really. The idea was that if `out` directory was _not_ deleted before running `vstart.sh`, rotating logs can separate current run of Ceph cluster from previous one.

Reply:
I'm not sure where it happens but my experience has been that starting a new cluster using `vstart.sh` always nukes the old logs. Is this not the case?

Reply:
I re-checked, I was under the wrong impression since vstart_runner.py deletes `out` and `dir` when it run `vstart.sh`.

Reply:
> Is it necessary to run this before hand?

Removed this line.

> I'm wondering if it makes sense to always produce a logrotate.conf|state. The logrotate process is not persistent so there's no reason to not create the conf file.

Produce everytime `vstart.sh` runs?